src/main/java/uk/gov/hmcts/reform/sendletter/tasks/UploadLettersTask.java

```diff
@@ -79,20 +84,25 @@ public void run() {
     }
 
     private void uploadToFtp(Letter letter) {
-        FileToSend file = new FileToSend(
-            FinalPackageFileNameHelper.generateName(letter),
-            letter.getFileContent()
-        );
-
-        ftp.upload(file, isSmokeTest(letter), letter.getService());
-
-        logger.info(
-            "Uploaded letter id: {}, messageId: {}, file name: {}, additional data: {}",
-            letter.getId(),
-            letter.getMessageId(),
-            file.filename,
-            letter.getAdditionalData()
-        );
+        Optional<String> serviceFolder = serviceFolderMapping.getFolderFor(letter.getService());
+        if (serviceFolder.isPresent()) {
```

Comment:
`ifPresent` could be used instead. Not much difference, up to you.

Reply:
it can't, I need an `else`. `ifPresent` returns void.

Reply:
true, what was I thinking?